### PR TITLE
Multi-contour per VOI

### LINKED ISSUE #465: Only one contour per slice
Pytrip returns only one contour per slice. If there are several contours per slice, pytrip shows only one. For example the skin is shown as one volume, instead of a layer. Or only one of the two lungs is shown.

Skin contour Eclipse:
skineclipse1_

Skin contour PyTRiPGUI:
skinpytripgui_dcm_

Lung contour Eclipse:
lungseclipse1_

Lung contour PyTRiPGUI:
lungspytripgui_dcm_

### PR DESCRIPTION
**Pull request by nbassler**
Closes #465 on PyTRiP side.

### CODE REVIEW

**Comment by nbassler** [comment]
Note this PR will break the API (also on GUI side).

@grzanka This may require some discussion. TRiP does not support multiple contours per VOI, so the concat() methods should be checked.

File: pytrip/vdx.py [hunk 1]
```diff
@@ -143,19 +147,15 @@ def read_dicom(self, data, structure_ids=None):
         self._structs_dicom_series_instance_uid = dcm.SeriesInstanceUID
         self._structs_sop_instance_uid = dcm.SOPInstanceUID
 
-        if hasattr(dcm, 'ROIContours'):
```
Comment (nbassler) [inline pytrip/vdx.py]: deprecated

File: pytrip/vdx.py [hunk 2]
```diff
@@ -143,19 +147,15 @@ def read_dicom(self, data, structure_ids=None):
...
-        elif hasattr(dcm, 'ROIContourSequence'):
+        if hasattr(dcm, 'ROIContourSequence'):
             _contours = dcm.ROIContourSequence
         else:
             logger.error("No ROIContours or ROIContourSequence found in dicom RTSTRUCT")
             exit()
 
         for i, item in enumerate(_contours):
             if structure_ids is None or item.RefdROINumber in structure_ids:
-                if hasattr(dcm, "RTROIObservations"):
```
Comment (nbassler) [inline pytrip/vdx.py]: deprecated

File: pytrip/vdx.py [hunk 3]
```diff
@@ -1329,45 +1359,46 @@ def number_of_contours(self):
...
+        return len(self.contours)
 
     def concat_contour(self):
         """ Concat all contours in this Slice object to a single contour.
         """
-        for i in range(len(self.contour) - 1, 0, -1):
-            self.contour[0].push(self.contour[i])
-            self.contour.pop(i)
-        self.contour[0].concat()
+        for i in range(len(self.contours) - 1, 0, -1):
```
Comment (grzanka) [inline pytrip/vdx.py]: I know it is out of the scope of this PR, but what is the aim of this loop ?
Looks like an error-prone section.

**Comment by grzanka** [comment]
@nbassler is there any way to test this PR ? Do you have some samples I could work with ?

**Comment by grzanka** [comment]
@nbassler to make test machinery happy, `pytest-timeout` should be removed from https://github.com/pytrip/pytrip/blob/master/tests/requirements-test.txt

File: pytrip/vdx.py (hunk 3 above)
Comment (nbassler) [reply pytrip/vdx.py]: It is something @jakobtoftegaard wrote, I am not sure either what it does. Some sort of reverse loop, concating all contours starting at the last contour in the list or so.
Also the child/children concept is obscure, why it is needed.

That concat() method should probably be refactored.

File: pytrip/vdx.py [hunk 4]
```diff
@@ -1026,17 +1032,37 @@ def read_dicom(self, info, data):
...
-                sl.add_dicom_contour(contour)
-                # TODO: check on con.ContourGeometricType = 'CLOSED_PLANAR'
-                # so far we simply assume all contours from DICOM are closed.
-                if sl.contour[-1].number_of_points() > 3:
-                    sl.contour[-1].contour_closed = True
-                else:
-                    sl.contour[-1].contour_closed = False
                 self.slices.append(sl)
+            else:
+                # lookup proper slice (just to be sure, should the contours come in random order)
```
Comment (nbassler) [inline pytrip/vdx.py]: @grzanka At this point in the code a marker could be added, this part of code is only executed if there are multiple contours for the same VOI in the same slice.

Then dicom2trip.py can be used to find datasets in slicerRT db which could be suitable for testing.

**Comment by nbassler** [comment]
@grzanka I only have a private dataset, but see previous comment regarding slicerRT

**Comment by nbassler** [comment]
Conversion to Single-Contours happens only when saving the VDX files. Maybe this is good enough for now. Concat of contours upon VDX save may be treated as a new issue (probably broken too).

**Comment by nbassler** [comment]
@grzanka 
pinnacle3-9.9-phantom-imrt 
ROI "Insert"

**Comment by nbassler** [comment]
DICOM sucks.
```
>>> dp = "/local/dicom/pinnacle3-9.9-phantom-imrt/"
>>> dcm = pt.dicomhelper.read_dicom_dir(dp)
>>> dcm['rtss'].ROIContourSequence[2].ContourSequence[16].NumberOfContourPoints
'1'
>>> dcm['rtss'].ROIContourSequence[2].ContourSequence[16].ContourData
['73.693695068359', '11.835327148438', '40']
>>> dcm['rtss'].ROIContourSequence[2].ContourSequence[16].ContourGeometricType
'CLOSED_PLANAR'
```

File: tests/requirements-test.txt [hunk 5]
```diff
@@ -2,7 +2,7 @@ versioneer<0.18
 # pytest v 3.0, released in 08.2016 doesn't work well with Python 3.2 and 3.3 (as in 11.2017)
 pytest<3.0 ; python_version >= '3.2' and python_version < '3.4' or os_name == 'nt' # py 3.2, 3.3 or Windows
 pytest ; (python_version < '3.0' or python_version >= '3.4') and os_name != 'nt' # py different than 3.2 or 3.3
-pytest-timeout
+# pytest-timeout
```
Comment (grzanka) [inline tests/requirements-test.txt]: This can be completely removed

**Pull request merged**